plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1212,13 +1221,33 @@ public void addColumn(ConnectorSession session, ConnectorTableHandle tableHandle
         try {
             long commitVersion = handle.getReadVersion() + 1;
 
+            ImmutableMap.Builder<String, Map<String, Object>> columnMetadata = ImmutableMap.builder();
+            columnMetadata.putAll(getColumnsMetadata(handle.getMetadataEntry()));
```

Comment:
Note here that a column may have a primitive type:

```
        {
            "metadata": {
                "delta.columnMapping.id": 1,
                "delta.columnMapping.physicalName": "col-2d56000f-9719-4765-8839-971ceca321c9"
            },
            "name": "col1",
            "nullable": true,
            "type": "integer"
        },
```

or a nested type 

```
        {
            "metadata": {
                "delta.columnMapping.id": 2,
                "delta.columnMapping.physicalName": "col-5f6af4d4-494c-4a36-a638-33b36156f293"
            },
            "name": "c2",
            "nullable": true,
            "type": {
                "type": "struct",
                "fields": [
                    {
                        "metadata": {
                            "delta.columnMapping.id": 3,
                            "delta.columnMapping.physicalName": "col-3347f394-1fcd-4a81-9ebf-95b73a5fe3af"
                        },
                        "name": "leaf1",
                        "nullable": true,
                        "type": "integer"
                    },
                    {
                        "metadata": {
                            "delta.columnMapping.id": 4,
                            "delta.columnMapping.physicalName": "col-eb56f195-96fc-40a8-9799-9178a673b91b"
                        },
                        "name": "leaf2",
                        "nullable": true,
                        "type": "integer"
                    }
                ]
            }
```

Retrieving the only the `metadata` of the existing _base_ element is not enough.


The following scenario corrupts the table:
```
spark-sql> create table tiny.test4 (c1 integer) using delta tblproperties ('delta.columnmapping.mode'='id');
```

```
trino> alter table delta.tiny.test4 add column c2 row(leaf1 integer, leaf2 integer);
ADD COLUMN
trino> alter table delta.tiny.test4 add column c3 row(leaf1 integer, leaf2 integer);
ADD COLUMN
```

Reply:
Ugh, I was aware of the issue before sending this PR, but forgot to fix it. Thanks for catching that.